crates/sui-rpc-benchmark/src/json_rpc/runner.rs

```diff
@@ -188,104 +188,212 @@ pub async fn run_queries(
+                    let now = Instant::now();
+                    debug!("Sending request for method: {}", request_line.method);
+                    let res = client
+                        .post(&endpoint)
+                        .json(&request_line.body_json)
+                        .send()
+                        .await;
+                    let elapsed_ms = now.elapsed().as_millis() as f64;
 
-                let now = Instant::now();
-                let res = client
-                    .post(&endpoint)
-                    .json(&request_line.body_json)
-                    .send()
-                    .await;
-                let elapsed_ms = now.elapsed().as_millis() as f64;
+                    // update pagination cursor if the request is successful.
+                    let mut is_error = true;
+                    if let Ok(resp) = res {
+                        if resp.status().is_success() {
+                            let supports_pagination = PaginationCursorState::get_method_cursor_index(&request_line.method).is_some();
+                            if supports_pagination {
+                                #[derive(Deserialize)]
+                                struct Body {
+                                    result: Result,
+                                }
+                                #[derive(Deserialize)]
+                                #[serde(rename_all = "camelCase")]
+                                struct Result {
+                                    has_next_page: bool,
+                                    next_cursor: Option<Value>,
+                                }
 
-                // update pagination cursor if the request is successful.
-                let mut is_error = true;
-                if let Ok(resp) = res {
-                    if resp.status().is_success() {
-                        #[derive(Deserialize)]
-                        struct Body {
-                            result: Result,
-                        }
-                        #[derive(Deserialize)]
-                        #[serde(rename_all = "camelCase")]
-                        struct Result {
-                            has_next_page: bool,
-                            next_cursor: Option<Value>,
+                                let resp_text = match resp.text().await {
+                                    Ok(text) => text,
+                                    Err(e) => {
+                                        return Err(BenchmarkError::Other(anyhow::anyhow!(
+                                            "Failed to get response text for method {}: {}", 
+                                            request_line.method, e
+                                        )));
+                                    }
+                                };
+                                let parse_result = serde_json::from_str::<Body>(&resp_text);
+                                if let Ok(Body { result }) = parse_result {
```

Comment:
Use the pattern match to help type inference figure out what is being parsed.
```suggestion
                                if let Ok(Body { result }) = serde_json::from_str(&resp_text) {
```

Reply:
the parse_result is also used below with `parse_result.err()`, hence the variable.